vortex-dtype/src/arbitrary.rs

```diff
@@ -11,15 +11,21 @@ impl<'a> Arbitrary<'a> for DType {
 }
 
 fn random_dtype(u: &mut Unstructured<'_>, depth: u8) -> Result<DType> {
-    let max_dtype_kind = if depth == 0 { 3 } else { 4 };
-    Ok(match u.int_in_range(0..=max_dtype_kind)? {
+    const BASE_TYPE_COUNT: i32 = 4;
+    const CONTAINER_TYPE_COUNT: i32 = 2;
+    let max_dtype_kind = if depth == 0 {
+        BASE_TYPE_COUNT
+    } else {
+        CONTAINER_TYPE_COUNT + BASE_TYPE_COUNT
+    };
+    Ok(match u.int_in_range(0..=(max_dtype_kind - 1))? {
         0 => DType::Bool(u.arbitrary()?),
         1 => DType::Primitive(u.arbitrary()?, u.arbitrary()?),
         2 => DType::Utf8(u.arbitrary()?),
         3 => DType::Binary(u.arbitrary()?),
         4 => DType::Struct(random_struct_dtype(u, depth - 1)?, u.arbitrary()?),
+        5 => DType::List(Arc::new(random_dtype(u, depth - 1)?), u.arbitrary()?),
```

Comment:
Would it be simpler if we kept struct as the last number? The logic became very complicated

Reply:
I can fold in the numbers, I preferred the explicitness. But the order wouldnt make anything simpler.

Reply:
I see, list is like struct. I was confused by seeing 4 since that includes Struct but then we do a range for the type - 1. I think it would be simpler to change `BASE_TYPE_COUNT` to 3 and remove `-1` from the int_in_range

Reply:
I can do that, ill rename BASE_TYPE_COUNT to base type offset